src/hotspot/share/compiler/directivesParser.cpp

```diff
@@ -336,12 +336,11 @@ bool DirectivesParser::set_option_flag(JSON_TYPE t, JSON_VAL* v, const key* opti
             error(VALUE_ERROR, "Unrecognized intrinsic detected in DisableIntrinsic: %s", validator.what());
           }
         } else if (strncmp(option_key->name, "PrintIdealPhase", 15) == 0) {
-          uint64_t mask = 0;
-          PhaseNameValidator validator(s, mask);
+          PhaseNameValidator validator(s);
 
           valid = validator.is_valid();
           if (valid) {
-            set->set_ideal_phase_mask(mask);
+            set->set_ideal_phase_mask(validator);
```

Comment:
Why not pass it the mask directly?

Reply:
Do you mean extracting the mask from the validator and then passing it to `set_ideal_phase_mask`? Isn't it simpler to just pass the validator and then handle it internally in `set_ideal_phase_mask`?

I've also renamed the above after Roberto's suggestion, so it's now:
```
set->set_ideal_phase_name_set(validator);
```

Reply:
Yes, extract the mask/set first, and pass it directly. There is no need for `set_ideal_phase_name_set` to get more than it asks for as input.

Reply:
OK, fine with me. Now changed.